### PR TITLE
Set local_error if handshake fails with SSL_ERROR_SSL

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We should set the local_error if the handshake fails with SSL_ERROR_SSL as we can't recover from it.

Modifications:

Set local_error when SSL_ERROR_SSL is received

Result:

Automatically close the connection on SSL_ERROR_SSL

### CODE REVIEW

File: quiche/src/tls.rs [hunk 1]
```diff
@@ -1093,6 +1095,24 @@ fn map_result_ptr<'a, T>(bssl_result: *const T) -> Result<&'a T> {
     }
 }
 
+fn set_local_error_if_needed(
```
Comment (tivrfoa) [inline quiche/src/tls.rs:1098]: Hi!

Why is this function defined outside of `impl Handshake`?

If it was an associated function (defined inside impl ...) you could then just call like:<br>
`self.set_local_error_if_needed(ex_data, rc);`

File: quiche/src/tls.rs [hunk 2]
```diff
@@ -1093,6 +1095,24 @@ fn map_result_ptr<'a, T>(bssl_result: *const T) -> Result<&'a T> {
...
+) {
+    // SSL_ERROR_SSL
+    if ssl.get_error(bssl_result) == 1 {
+        // SSL_ERROR_SSL can't be recovered so ensure we set a
+        // local_error so the connection is closed.
+        // See https://www.openssl.org/docs/man1.1.1/man3/SSL_get_error.html
+        if ex_data.local_error.is_none() {
+            *ex_data.local_error = Some(ConnectionError {
+                is_app: false,
+                error_code: TLS_ALERT_ERROR,
```
Comment (ghedo) [inline quiche/src/tls.rs:1109]: This  is not actually a TLS alert though, so it should be `0x01` (`INTERNAL_ERROR`) instead IMO.

File: quiche/src/tls.rs (hunk 1 above)
Comment (ghedo) [reply quiche/src/tls.rs:1098]: Yeah, better make this a method of `Handshake` (`map_result_ssl()` should probably be too, but that's a separate issue).

Also, maybe call this `set_transport_error()` or something.

**Review by ghedo — CHANGES_REQUESTED** [review]

**Comment by normanmaurer** [comment]
@ghedo I can open a PR to move `map_result_ssl ` as well

**Comment by normanmaurer** [comment]
@ghedo FYI https://github.com/cloudflare/quiche/pull/1342

**Review by sildvauresberg — DISMISSED** [review]

**Comment by normanmaurer** [comment]
@ghedo should I rebase now that https://github.com/cloudflare/quiche/pull/1342 is merged ?

**Comment by ghedo** [comment]
@normanmaurer Yeah, or I'll do it in a bit.

**Comment by normanmaurer** [comment]
@ghedo done

**Review by ghedo — APPROVED** [review]

**Pull request merged**